packages/core/src/browser/quick-input/quick-input-service.ts

```diff
@@ -207,13 +208,86 @@ export interface QuickInputService {
     hide(): void;
 }
 
+/**
+ * Filter the list of quick pick items based on the provided filter.
+ * Items are filtered based on if:
+ * - their `label` satisfies the filter using `fuzzy`.
+ * - their `description` satisfies the filter using `fuzzy`.
+ * - their `detail` satisfies the filter using `fuzzy`.
+ * Filtered items are also updated to display proper highlights based on how they were filtered.
+ *
+ * @param items the list of quick pick items.
+ * @param filter the filter to search for.
+ *
+ * @returns the list of quick pick items that satisfy the filter.
+ */
 export function filterItems(items: QuickPickItem[], filter: string): QuickPickItem[] {
-    return filter.trim().length === 0 ? items : items
-        .filter(item => item.label.toLowerCase().indexOf(filter.toLowerCase()) > -1)
-        .map(item => Object.assign(item, { highlights: { label: findMatches(item.label.toLowerCase(), filter.toLowerCase()) } }));
+    filter = filter.trim().toLowerCase();
+
+    if (filter.length === 0) {
+        for (const item of items) {
+            item.highlights = undefined; // reset highlights from previous filtering.
+        }
+        return items;
+    }
+
+    const filteredItems: QuickPickItem[] = [];
+    for (const item of items) {
+        if (fuzzy.test(filter, item.label.toLowerCase())) {
+            item.highlights = { label: findMatches(item.label, filter) };
+            filteredItems.push(item);
+        } else if (item.description && fuzzy.test(filter, item.description)) {
+            item.highlights = { description: findMatches(item.description, filter) };
+            filteredItems.push(item);
+        } else if (item.detail && fuzzy.test(filter, item.detail)) {
+            item.highlights = { detail: findMatches(item.detail, filter) };
+            filteredItems.push(item);
```

Comment:
These highlightings are exclusive, so highlights for `{ label: 'test', description: 'test' }` when searching for `test` will only highlight the label. Compare this with the behavior of vscode, where both are highlighted:

![grafik](https://user-images.githubusercontent.com/4377073/129913548-b8ade445-0fbc-4fdc-a879-7242697d003e.png)